plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcFileWriterFactory.java

```diff
@@ -168,12 +174,14 @@ public Optional<FileWriter> createFileWriter(
                 validationInputFactory = Optional.of(() -> {
                     try {
                         TrinoInputFile inputFile = fileSystem.newInputFile(stringPath);
-                        return new HdfsOrcDataSource(
+                        HdfsOrcDataSource hdfsOrcDataSource = new HdfsOrcDataSource(
                                 new OrcDataSourceId(stringPath),
                                 inputFile.length(),
                                 new OrcReaderOptions(),
                                 inputFile,
                                 readStats);
+                        orcFileOperationStats.getOpenStats().addNanos(hdfsOrcDataSource.getFileOpenTime());
```

Comment:
in addition to Raunaq's comment, the tracking part can be contained within Trino's FS abstractions - and don't need to be exposed outside.